### PR TITLE
T13544-part2

### PR DESCRIPTION
**Pull request by tchx84**
https://phabricator.endlessm.com/T13544
https://bugzilla.gnome.org/show_bug.cgi?id=773747

### CODE REVIEW

File: gtk/gtkfilechooserwidget.c [hunk 1]
```diff
@@ -6180,15 +6180,32 @@ gtk_file_chooser_widget_get_default_size (GtkFileChooserEmbed *chooser_embed,
...
+  window = gtk_widget_get_window (GTK_WIDGET (impl));
+  if (screen && window)
+    {
+      monitor_num = gdk_screen_get_monitor_at_window (screen, window);
+      gdk_screen_get_monitor_workarea (screen, monitor_num, &monitor);
+      max_width = monitor.width * 5 / 6;
+      max_height = monitor.height * 5 / 6;
+    }
+  else
+    max_width = max_height = G_MAXINT;
```
Comment (xjuan) [inline gtk/gtkfilechooserwidget.c]: nit: you could move this block to a function _gtk_file_chooser_get_monitor_workarea_for_widget()

File: gtk/gtkfilechooserwidget.c [hunk 2]
```diff
@@ -6180,15 +6180,32 @@ gtk_file_chooser_widget_get_default_size (GtkFileChooserEmbed *chooser_embed,
...
+
+  screen = gtk_widget_get_screen (GTK_WIDGET (impl));
+  window = gtk_widget_get_window (GTK_WIDGET (impl));
+  if (screen && window)
+    {
+      monitor_num = gdk_screen_get_monitor_at_window (screen, window);
+      gdk_screen_get_monitor_workarea (screen, monitor_num, &monitor);
+      max_width = monitor.width * 5 / 6;
+      max_height = monitor.height * 5 / 6;
+    }
```
Comment (xjuan) [inline gtk/gtkfilechooserwidget.c]: nit: you could move this block to a function _gtk_file_chooser_get_monitor_workarea_for_widget()

File: gtk/gtkfilechooserwidget.c [hunk 3]
```diff
@@ -6180,15 +6180,32 @@ gtk_file_chooser_widget_get_default_size (GtkFileChooserEmbed *chooser_embed,
...
-  int x, y, width, height;
+  int x, y, width, height, max_width, max_height;
   GSettings *settings;
+  GdkScreen *screen;
+  GdkWindow *window;
+  gint monitor_num;
+  GdkRectangle monitor;
+
+  screen = gtk_widget_get_screen (GTK_WIDGET (impl));
+  window = gtk_widget_get_window (GTK_WIDGET (impl));
```
Comment (xjuan) [inline gtk/gtkfilechooserwidget.c]: You could use gdk_screen_get_root_window() will wont fail if the widget is not realized yet

File: gtk/gtkfilechooserwidget.c (hunk 2 above)
Comment (tchx84) [inline gtk/gtkfilechooserwidget.c]: Thanks for explaining, will move it there.

File: gtk/gtkfilechooserwidget.c [hunk 4]
```diff
@@ -6173,22 +6173,52 @@ find_good_size_from_style (GtkWidget *widget,
...
+  screen = gtk_widget_get_screen (widget);
+  if (!screen)
+    {
+      *width = 1920;
+      *height = 1080;
+      return;
+    }
+
+  window = gdk_screen_get_root_window (screen);
+  monitor_num = gdk_screen_get_monitor_at_window (screen, window);
```
Comment (xjuan) [inline gtk/gtkfilechooserwidget.c]: Sorry, my bad, you need to use the widget window for this to work properly

**Comment by tchx84** [comment]
I moving this commit to the main PR, so is easier to review all together.

**Pull request closed without merging**